src/include/packager/repositories_include.rb

```diff
@@ -25,329 +34,179 @@ def initialize_packager_repositories_include(_include_target)
-                    "Change the protocol or unpack the ISO image on the server side."
-                )
-              )
-            end
-          end
+        # for local repositories (e.g. CD/DVD) which have autorefresh disabled
+        # download the metadata immediately, the medium is in the drive right
+        # now, it can be changed later and accidentally added a different repository
+        if !autorefresh_for?(url)
+          log.info "Adding a local repository, refreshing it now..."
+          Pkg.SourceRefreshNow(new_repo_id)
+        end
+      end
 
-          # popup message part 2
-          msg = Ops.add(
-            Ops.add(msg, "\n\n"),
-            _("Change the URL and try again?")
-          )
+      # broken repository or wrong URL - enter the URL again
+      if enter_again
+        Pkg.SourceReleaseAll
+        return :again
+      end
 
-          tryagain = Popup.YesNo(msg)
-          return :again if tryagain
+      log.info("New sources: #{newSources}")
 
-          return :cancel
-        else
-          Progress.NextStage
-          license_accepted = true
-          Builtins.foreach(newSources) do |id|
-            if !LicenseAccepted(id)
-              Builtins.y2milestone("License NOT accepted, removing the source")
-              Pkg.SourceDelete(id)
-              license_accepted = false
-            else
-              src_data = Pkg.SourceGeneralData(id)
-              Builtins.y2milestone("Addded repository: %1", src_data)
-
-              sourceState = {
-                "SrcId"       => id,
-                "enabled"     => Ops.get_boolean(src_data, "enabled", true),
-                "autorefresh" => Ops.get_boolean(src_data, "autorefresh", true),
-                "name"        => Ops.get_string(src_data, "name", ""),
-                "do_refresh"  => download
-              }
-              @sourceStatesOut = Builtins.add(@sourceStatesOut, sourceState)
-            end
+      if newSources.empty?
+        log.error("Cannot add the repository")
+        try_again(url, scheme) ? :again : :cancel
+      else
+        Progress.NextStage
+        license_accepted = true
+        Builtins.foreach(newSources) do |id|
+          if !LicenseAccepted(id)
+            log.info("License NOT accepted, removing the source")
+            Pkg.SourceDelete(id)
+            license_accepted = false
+          else
+            src_data = Pkg.SourceGeneralData(id)
+            log.info("Addded repository: #{src_data}")
+
+            sourceState = {
+              "SrcId"       => id,
+              "enabled"     => src_data["enabled"],
+              "autorefresh" => src_data["autorefresh"],
+              "name"        => src_data["name"],
+              "do_refresh"  => download
+            }
+            @sourceStatesOut << sourceState
           end
+        end
 
-          # relese (unmount) the medium
-          Pkg.SourceReleaseAll
+        # relese (unmount) the medium
+        Pkg.SourceReleaseAll
```

Comment:
well, I expect that this can be in `ensure` and in such case work-flow in this method can be simplified by `return :abort` on line 184